agent/consul/leader_peering.go

```diff
@@ -87,11 +94,11 @@ func (s *Server) emitPeeringMetricsOnce(logger hclog.Logger, metricsImpl *metric
 	for _, peer := range peers {
 		status, found := s.peerStreamServer.StreamStatus(peer.ID)
 		if !found {
-			logger.Trace("did not find status for", "peer_name", peer.Name)
+			logger.Trace("did not find status for; will skip metric emission",
```

Comment:
I think we should ensure ever peering emits a metric. In this case missing the peer status should be a signal that something is wrong so it should be pretty safe to mark it unhealthy.

Reply:
Did you mean for the `health` metric specifically or all metrics here? Like `exported services counts`